src/uwtools/config/tools.py

```diff
@@ -172,26 +162,29 @@ def _print_config_section(config: dict, key_path: List[str]) -> None:
 
 def _realize_config_update(
     config_obj: Config,
+    config_fmt: str,
     supplemental_configs: Optional[List[Union[dict, Config, DefinitePath]]] = None,
 ) -> Config:
     """
     Update config with values from other configs, if given.
 
     :param config_obj: The config to update.
+    :param config_fmt: Format of config's source.
     :param supplemental_configs: Sources of values to modify input.
     :return: The input config, possibly updated.
     """
     if supplemental_configs:
         log.debug("Before update, config has depth %s", config_obj.depth)
         supplemental_obj: Config
-        for config in supplemental_configs:
-            if isinstance(config, dict):
-                supplemental_obj = YAMLConfig(config=config)
-            elif isinstance(config, Config):
-                supplemental_obj = config
+        for idx, supplemental_config in enumerate(supplemental_configs):
+            _validate_format_supplemental(config_fmt, supplemental_config, idx)
```

Comment:
This call validates that the format of supplemental configs matches the input format.